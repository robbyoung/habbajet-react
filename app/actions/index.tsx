export * from './actionTypes';
export * from './addHabbajet';
export * from './addHabitResult';
export * from './updateBudget';
export * from './resetHabbajet';
export * from './loadState';
export * from './selectHabbajet';
export * from './updateEditorField';
export * from './validateEditor';
export * from './clearEditor';
