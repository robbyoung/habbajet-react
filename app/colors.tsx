export const white = '#ffffff';
export const lightGrey = 'F5F5F5';
export const grey = '#959595';

export const habbajetColors = [
    '#8066C2', // Purple
    '#2AAD5E', // Green
    '#E04747', // Red
    '#7F90EC', // Blue
    '#B1B952', // Gold
];

export const complementColors: {[id: string]: string} = {
    '#8066C2': '#C2A066',
    '#2AAD5E': '#AD372A',
    '#E04747': '#47E047',
    '#7F90EC': '#ECB27F',
    '#B1B952': '#B952B0',
};
