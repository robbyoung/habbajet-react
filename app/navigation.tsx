import {Navigation} from 'react-native-navigation';
import React from 'react';
import HabbajetScreen from './containers/habbajetScreen';
import LoadingScreen from './containers/loadingScreen';
import {Provider} from 'react-redux';
import store from './store';

enum Screens {
    Loading = 'Loading',
    Habbajet = 'Habbajet',
}

export const STACK_NAVIGATOR = 'StackNavigator';

Navigation.registerComponent(
    Screens.Loading,
    () => () => (
        <Provider store={store}>
            <LoadingScreen />
        </Provider>
    ),
    () => LoadingScreen,
);

Navigation.registerComponent(
    Screens.Habbajet,
    () => () => (
        <Provider store={store}>
            <HabbajetScreen />
        </Provider>
    ),
    () => HabbajetScreen,
);

export const goToLoading = () => {
    Navigation.setRoot({
        root: {
            stack: {
                id: STACK_NAVIGATOR,
                children: [
                    {
                        component: {
                            name: Screens.Loading,
                            options: {
                                topBar: {
                                    visible: false,
                                },
                            },
                        },
                    },
                ],
            },
        },
    });
};

export const goToHabbajet = () => {
    Navigation.setRoot({
        root: {
            stack: {
                id: STACK_NAVIGATOR,
                children: [
                    {
                        component: {
                            name: Screens.Habbajet,
                            options: {
                                topBar: {
                                    title: {
                                        text: ' ',
                                        fontFamily: 'Abel',
                                        fontSize: 30,
                                        color: '#ffffff',
                                    },
                                },
                            },
                        },
                    },
                ],
            },
        },
    });
};
