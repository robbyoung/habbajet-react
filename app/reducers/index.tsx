import habbajets from './habbajets';
import purchases from './purchases';
import budget from './budget';
import habbajetEditor from './habbajetEditor';
import purchaseEditor from './purchaseEditor';
import tagEditor from './tagEditor';
import tags from './tags';
import {combineReducers} from 'redux';

export default combineReducers({
    habbajets,
    purchases,
    budget,
    tags,
    habbajetEditor,
    purchaseEditor,
    tagEditor,
});
