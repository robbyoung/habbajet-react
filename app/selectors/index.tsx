export * from './budget';
export * from './purchases';
export * from './habbajets';
export * from './habbajetEditor';
