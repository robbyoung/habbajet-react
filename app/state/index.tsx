export enum HabitResult {
    Success,
    SlackSuccess,
    Failure,
}

export interface Habbajet {
    id: string;
    name: string;
    maxValue: number;
    currentValue: number;
    modifier: number;
    results: HabitResult[];
    color: string;
    date: string;
    toClaim: boolean;
    selected: boolean;
    currentStreak: number;
    bestStreak: number;
    oldStreaks: [number, number];
    totalSlack: number;
    remainingSlack: number;
}

export interface Purchase {
    id: string;
    name: string;
    cost: number;
    date: string;
    tagId: string;
}

export interface EditorField {
    value: string;
    errorMessage: string;
}

export interface HabbajetEditor {
    name: EditorField;
    value: EditorField;
    modifier: EditorField;
    slack: EditorField;
    color: string;
    validated: boolean;
}

export interface PurchaseEditor {
    name: EditorField;
    cost: EditorField;
    tagId: string;
    validated: boolean;
    id?: string;
}

export interface Tag {
    id: string;
    name: string;
    color: string;
}

export interface State {
    habbajets: Habbajet[];
    purchases: Purchase[];
    tags: Tag[];
    habbajetEditor: HabbajetEditor;
    purchaseEditor: PurchaseEditor;
    budget: number;
}
