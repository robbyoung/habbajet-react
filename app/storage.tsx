import AsyncStorage from '@react-native-community/async-storage';
import {State} from './state';
import store from './store';
import {createTestState} from './state/testState';

// const defaultState: State = {
//     budget: 0,
//     habbajets: [],
//     purchases: [],
// };

const defaultState = createTestState(1, 0, 0);

export async function saveState(): Promise<void> {
    const state = store.getState();
    await AsyncStorage.setItem('state', JSON.stringify(state));
}

export async function loadState(): Promise<State> {
    const state = await AsyncStorage.getItem('state');
    let savedState: State;
    if (state !== undefined && state !== null) {
        savedState = JSON.parse(state) as State;
        return savedState;
    }
    return defaultState;
}

export async function clearState() {
    await AsyncStorage.clear();
}
