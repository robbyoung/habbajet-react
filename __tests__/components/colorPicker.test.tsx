import React from 'react';
import renderer from 'react-test-renderer';
import ColorPicker from '../../app/components/colorPicker';
import {habbajetColors} from '../../app/colors';
import {fireEvent, render, wait} from '@testing-library/react-native';

describe('ColorPicker Component', () => {
    it('can render a color picker', () => {
        const component = renderer.create(
            <ColorPicker
                selected={habbajetColors[0]}
                onSelect={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will run the onSelect event if a color is selected', async () => {
        const onSelect = jest.fn();
        const {getByTestId} = render(
            <ColorPicker selected={habbajetColors[0]} onSelect={onSelect} />,
        );

        const colorToSelect = getByTestId(`color-${habbajetColors[1]}`);
        fireEvent.press(colorToSelect);
        await wait(() => expect(onSelect).toBeCalledWith(habbajetColors[1]));
    });
});
