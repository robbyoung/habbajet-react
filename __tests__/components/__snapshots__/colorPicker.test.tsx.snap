// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Component can render a color picker 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#959595",
          "fontFamily": "Abel",
          "fontSize": 20,
          "margin": 0,
          "padding": 0,
        },
      ]
    }
  >
    Color
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#8066C2",
          "borderColor": "#959595",
          "borderRadius": 4,
          "borderStyle": "dotted",
          "borderWidth": 4,
          "height": 60,
          "marginBottom": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="color-#8066C2"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E04747",
          "borderRadius": 4,
          "height": 60,
          "marginBottom": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="color-#E04747"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2AAD5E",
          "borderRadius": 4,
          "height": 60,
          "marginBottom": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="color-#2AAD5E"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#B1B952",
          "borderRadius": 4,
          "height": 60,
          "marginBottom": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="color-#B1B952"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#7F90EC",
          "borderRadius": 4,
          "height": 60,
          "marginBottom": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="color-#7F90EC"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#37353C",
          "borderRadius": 4,
          "height": 60,
          "marginBottom": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="color-#37353C"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E64CE9",
          "borderRadius": 4,
          "height": 60,
          "marginBottom": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="color-#E64CE9"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#EB712D",
          "borderRadius": 4,
          "height": 60,
          "marginBottom": 10,
          "marginRight": 10,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="color-#EB712D"
    />
  </View>
</View>
`;
